From 3d102676de8944fdad4cc27e7828040f4bdf3cd5 Mon Sep 17 00:00:00 2001
From: Stefan Metzmacher <metze@samba.org>
Date: Mon, 12 Jul 2010 14:05:51 +0200
Subject: [PATCH 10/12] config/interface_modify.sh: before calling a script check if it exists and is executable

For non bash shells $_s_script might end with '/*'.

We do the workarround this way, because it makes sense to check
that a script is executable, before trying to execute it.

metze

[ This actually applies to any shell -- Rusty Russell ]
Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
---
 config/interface_modify.sh |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/config/interface_modify.sh b/config/interface_modify.sh
index 738530b..69601dd 100755
--- a/config/interface_modify.sh
+++ b/config/interface_modify.sh
@@ -68,6 +68,9 @@ delete_ip_from_iface()
 
 		local _s_script=""
 		for _s_script in $_s_script_dir/*; do
+			test -x "$_s_script" || {
+				continue
+			}
 			$_s_script "$_iface" "$_s_ip" "$_s_maskbits" || {
 				ret=$?
 				echo "$_s_script '$_iface' '$_s_ip' '$_s_maskbits' - failed - $ret"
-- 
1.7.4

